lib/bots/versionbot.ts

```diff
@@ -212,11 +212,38 @@ export class VersionBot extends GithubBot.GithubBot {
             // Go through all the commits. We're looking for, at a minimum, a 'change-type:' tag.
             for (let commit of commits) {
                 const commitMessage: string = commit.commit.message;
-                const invalidCommit = !commitMessage.match(/^change-type:\s*(patch|minor|major)\s*$/mi);
 
-                if (!invalidCommit) {
-                    changetypeFound = true;
-                    break;
+                // Split the commits up into lines, and find the last line with any whitespace in.
+                // Whilst we tend to ask for:
+                //  <header>
+                //
+                //  <body>
+                //
+                //  <footer>
+                // This code will actually let you get away with:
+                //  <header>
+                //
+                //  <footer>
+                // As sometimes a development patch may be self-explanatory in the header alone.
+                const lines = commitMessage.split('\n');
+                let index = 0;
+                for (index = lines.length - 1; index >= 0; index -= 1) {
+                    if (lines[index].match(/$\s*^/)) {
```

Comment:
> A good catch.. yet it's worked with all the tests I've thrown at it. Which seems odd.

Ah no, got it! It _is_ possible for `/$\s*^/` to match, but only against the empty string, because then you're matching against `^` and `$` at the same time in the same place, with no spaces. This regex can still work, but only as long as it's on an actually empty line. It's just the whitespace-allowing part that will never match. Neat.